server/lib/bid/contracts/Bid.sol

```diff
@@ -52,4 +52,17 @@ contract Bid is ErrorCodes, BidState {
     supplierAddress.send(amountWei-fee);
     return ErrorCodes.SUCCESS;
   }
+
+  function reject(address buyer) returns (ErrorCodes) {
+    // confirm balance, to return error
+    if (this.balance < amount) {
+      return ErrorCodes.INSUFFICIENT_BALANCE;
+    }
+    uint fee = 10000000 wei; // buyer absorbs the fee
+    uint amountWei = amount * 1 ether;
+
+    // transfer will throw
+    buyer.send(amountWei-fee);
```

Comment:
@liorsaar1 the comment below regards this line. For some reason this transfer is not executing and the balance is not updating. I should be done with the task but wanted to get your opinion on if this is incorrect.